alpa/device_mesh.py

```diff
@@ -498,6 +513,93 @@ def run_allgather_task(self, uuid, buffer_uuids):
                            allgather_spec.tensor_slices,
                            allgather_spec.output_slice)
 
+    def broadcast(self, uuids, comm_key, world_size, devices_ids, devices_global_rank, tensor_slices, group_name):
+        used_tensors = []
+        slice_types = []            
+        is_bool = self.buffers[uuids[devices_ids[0]]].dtype == np.bool_
+        for device_id, global_rank, tensor_slice in zip(devices_ids, devices_global_rank, tensor_slices):
+            uuid = uuids[device_id]
+            tensor_shape = self.buffers[uuid].shape
+            slice_shape = tuple(ind.stop - ind.start for ind in tensor_slice)
+            if is_continuous_subset(tensor_slice, tensor_shape):
+                # fast path, two cases: (1) same shape, (2) continuous subset.
+                if slice_shape != tensor_shape:
+                    ind, _ = infer_offset_and_n_elements(tensor_slice)
+                    slice_types.append("continuous_subset")
+                    to_use = (xla_buffer_to_cupy(self.buffers[uuid]), ind)
```

Comment:
two types of `to_use` make the code more complicated(more `if` like L549 and L560, `slice_types`). Is it possible to use `to_use`(buffer with offset, for `col.broadcast`) and `buffer`(buffer without offset, for `cupy_to_xla`).

Reply:
Sounds like a good idea, thx!